web_generator/lib/src/ast/base.dart

```diff
@@ -147,3 +145,7 @@ class ParameterDeclaration {
       ..type = type.emit(TypeOptions(nullable: optional)));
   }
 }
+
+abstract class NamedType extends Type {
```

Comment:
Awesome! I didn't expect there would be such an elegant fix to the `null` `name`s.